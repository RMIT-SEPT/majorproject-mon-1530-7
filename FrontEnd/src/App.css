.App {
  text-align: center;
  background-color: rgba(164, 214, 197, 0.904);
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.page-container{
  display: flex;
  min-height: 100vh;
  flex-direction: column;
}

.content-wrap{
  flex:1;
}