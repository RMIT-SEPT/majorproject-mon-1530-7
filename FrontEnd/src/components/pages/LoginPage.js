import React, { Component } from 'react'
import Form from 'react-bootstrap/Form';
import Button from 'react-bootstrap/Button'
import UserProfile from '../../UserProfile.js'



class LoginPage extends Component {
    constructor(props) {
        super(props);
        this.state = {}
      }
    
      handleChange = (event) => {
        this.setState({[event.target.name]: event.target.value});
        
      }
    
      handleSubmit = (event) => {
        var username = this.state.username
        fetch(process.env["REACT_APP_API_URL"] + "/user/login", {
          method: "POST",
          headers: {
            "Content-Type": "application/json",
          },

          body: JSON.stringify({
            username: this.state.username,
            password: this.state.password,
          }),
        })
          .then(function (response) {
            if (!response.ok) {
              console.log("reject");
              throw new Error("Request failed");
            }

            return response.json();
          })
          .then((data) => {
            UserProfile.setLoggedIn();
            UserProfile.setAdmin(data.admin);
            UserProfile.setUID(data.userId);
            UserProfile.setToken(data.token);

            window.location.reload(false);
          })
          .catch((error) => alert("incorrect username or password"));

        event.preventDefault();
        
    }
    
    
    render() {
        
        
        return (
            <div className="loginContainer">
                <Form onSubmit={this.handleSubmit}>
                    <Form.Group controlId="formUsername">
                        <Form.Label>Username</Form.Label>
                        <Form.Control name="username" value={this.state.value} onChange={this.handleChange} type="username"  placeholder="Enter Username"/>
                    </Form.Group>
                    <Form.Group controlId="formPassword">
                        <Form.Label>Password</Form.Label>
                        <Form.Control name="password" value={this.state.value} onChange={this.handleChange} type="password" placeholder="Enter Password" />
                    </Form.Group>
                    <Button variant="primary" type="submit">
                        Login
                    </Button>
                </Form>
            </div>
        )
    }
}

export default LoginPage;
