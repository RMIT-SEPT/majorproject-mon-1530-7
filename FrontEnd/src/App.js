import React from "react";
import { Redirect } from 'react-router';
import NavigationBar from "./components/layouts/Navbar.js";
import Footer from "./components/layouts/Footer.js";
import { Switch, Route, BrowserRouter } from "react-router-dom";
import LoginPage from "./components/pages/LoginPage";
import HomePage from "./components/pages/HomePage";
import AboutPage from "./components/pages/AboutPage";
import ContactPage from "./components/pages/ContactPage";
import CustomerBookingPage from "./components/pages/CustomerBookingPage.js";
import NewEmployeePage from "./components/pages/NewEmployeePage.js";

import AdminPastBooking from "./components/pages/AdminPastBooking.js";
import CustomerPastBookingsPage from "./components/pages/CustomerPastBookingsPage.js";

import Dashboard from "./components/pages/Dashboard.js";
import AdminDashboard from "./components/pages/AdminDashboard.js";
import Account from "./components/pages/Account.js";
import Profile from "./components/pages/Profile.js";

import ManageEmp from "./components/pages/ManageEmp.js";
import EmpDetails from "./components/pages/EmpDetails.js";

import UserProfile from "./UserProfile.js"
import AdminPastBookingsPage from "./components/pages/AdminPastBookingsPage.js";

function App() {
  console.log(UserProfile.getLoggedIn())
  if(UserProfile.getLoggedIn()) {
    return (
      <div className="page-container">
        <div className="content-wrap">
          <BrowserRouter>
            <NavigationBar loggedIn={UserProfile.getToken()} />
            <Switch>
                <Route exact path="/" component={HomePage} />,
                <Redirect from="/login" to="dashboard" />,
                <Route path="/about" component={AboutPage} />,
                <Route path="/contact" component={ContactPage} />,
                <Route path="/booking" component={CustomerBookingPage} />,
                <Route
                  path="/customer-past-bookings"
                  component={CustomerPastBookingsPage}
                />,
                <Route path="/dashboard" component={Dashboard} >
                  {UserProfile.getAdmin() ? <Redirect to="/admin-dashboard" /> : <Dashboard />}
                </Route>,
                <Route path="/admin-dashboard" component={AdminDashboard} >
                  {UserProfile.getAdmin() ?   <AdminDashboard /> : <Redirect to="/dashboard" /> }
                </Route>,
                <Route path="/account" component={Account} >
                  {UserProfile.getLoggedIn() ? <Account /> : <Redirect to="/login"/>}
                </Route>,
                <Route path="/new-employee" component={NewEmployeePage} />,
                <Route path="/profile" component={Profile} />,
                <Route path="/manage-emp" component={ManageEmp} />,
                <Route path="/emp-details" component={EmpDetails} /> 
          
            </Switch>
          </BrowserRouter>
        </div>
        <Footer />
      </div>
    );
  }
  else {
    return (
      <div className="page-container">
        <div className="content-wrap">
          <BrowserRouter>
            <NavigationBar loggedIn={UserProfile.getToken()} />
            <Switch>
              
                <Route exact path="/" component={HomePage} />,
                <Route path="/login" component={LoginPage} />,
                <Route path="/about" component={AboutPage} />,
                <Route path="/contact" component={ContactPage} />,
                <Route path="/booking" component={CustomerBookingPage} />,
                <Route
                  path="/customer-past-bookings"
                  component={CustomerPastBookingsPage}
                />,
                <Redirect from="/dashboard" to="/login" />,
                <Route path="/admin-dashboard" component={AdminDashboard} >
                  {UserProfile.getAdmin() ?   <AdminDashboard /> : <Redirect to="/dashboard" /> }
                </Route>,
                <Redirect from="/account" to="/login" />,
                <Route path="/new-employee" component={NewEmployeePage} />,
                <Route path="/profile" component={Profile} />,
                <Route path="/manage-emp" component={ManageEmp} />,
                <Route path="/emp-details" component={EmpDetails} />  
            </Switch>
          </BrowserRouter>
        </div>
        <Footer />
      </div>
    );
  }
}
 

export default App;
