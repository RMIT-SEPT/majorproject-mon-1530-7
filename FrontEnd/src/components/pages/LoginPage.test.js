import React from 'react';
import {MemoryRouter} from 'react-router';
import LoginPage from './LoginPage';
import App from '../../App';
import {mount} from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('Testing LoginPage route', () => {
    it('should show LoginPage component for /login router (using memory router)', () => {
      const component = mount( <MemoryRouter initialEntries={["/login"]}>
          <App/>
       </MemoryRouter>
      );
      expect(component.contains(LoginPage));
    })
  })
