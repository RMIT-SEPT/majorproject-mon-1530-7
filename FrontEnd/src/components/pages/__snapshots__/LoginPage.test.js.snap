// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing LoginPage route renders correctly 1`] = `
<div
  className="login-container"
>
  <Form
    inline={false}
    onSubmit={[Function]}
  >
    <FormGroup
      controlId="formUsername"
      id="form-layout"
    >
      <FormLabel
        column={false}
        id="custom-form-label"
        srOnly={false}
      >
        Username
      </FormLabel>
      <FormControl
        name="username"
        onChange={[Function]}
        placeholder="Enter Username"
        type="username"
      />
    </FormGroup>
    <FormGroup
      controlId="formPassword"
      id="form-layout"
    >
      <FormLabel
        column={false}
        id="custom-form-label"
        srOnly={false}
      >
        Password
      </FormLabel>
      <FormControl
        name="password"
        onChange={[Function]}
        placeholder="Enter Password"
        type="password"
      />
    </FormGroup>
    <button
      className="btn-filled-alt"
    >
      Log In
    </button>
  </Form>
</div>
`;
